ssl-config/src/main/java/com/palantir/remoting1/config/ssl/SslSocketFactories.java

```diff
@@ -83,6 +83,21 @@ public static SSLContext createSslContext(SslConfiguration config) {
         return createTrustManagerFactory(config.trustStorePath(), config.trustStoreType()).getTrustManagers();
     }
 
+    /**
+     * Returns the first {@link TrustManager} initialized from the given configuration. This is always an {@link
+     * javax.net.ssl.X509TrustManager}.
+     */
+    public static X509TrustManager createX509TrustManager(SslConfiguration config) {
+        TrustManager trustManager = createTrustManagers(config)[0];
+        if (trustManager instanceof X509TrustManager) {
+            return (X509TrustManager) trustManager;
+        } else {
+            throw new RuntimeException(
+                    "First TrustManager associated with SslConfiguration was not a X509TrustManager: "
+                            + config.trustStorePath());
```

Comment:
done

On Wed, Jul 27, 2016 at 1:49 PM Mark Elliot <notifications@github.com>
wrote:

> In
> ssl-config/src/main/java/com/palantir/remoting1/config/ssl/SslSocketFactories.java
> <https://github.com/palantir/http-remoting/pull/188#discussion_r72520412>:
>
> > @@ -83,6 +83,21 @@ public static SSLContext createSslContext(SslConfiguration config) {
> >          return createTrustManagerFactory(config.trustStorePath(), config.trustStoreType()).getTrustManagers();
> >      }
> >
> > +    /**
> > +     * Returns the first {@link TrustManager} initialized from the given configuration. This is always an {@link
> > +     * javax.net.ssl.X509TrustManager}.
> > +     */
> > +    public static X509TrustManager createX509TrustManager(SslConfiguration config) {
> > +        TrustManager trustManager = createTrustManagers(config)[0];
> > +        if (trustManager instanceof X509TrustManager) {
> > +            return (X509TrustManager) trustManager;
> > +        } else {
> > +            throw new RuntimeException(
> > +                    "First TrustManager associated with SslConfiguration was not a X509TrustManager: "
> > +                            + config.trustStorePath());
>
> probably useful here to say what kind of trust manager it was?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/188/files/4a9584137fec93d0845e2cfb83e561dad9617d47#r72520412>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwd2dIEgum6qmEiVfb1N2wEQbHDRHks5qZ8QRgaJpZM4JWnD7>
> .
>